lib/adapters/xhr.js

```diff
@@ -109,7 +109,11 @@ module.exports = function xhrAdapter(resolve, reject, config) {
         undefined;
 
     if (xsrfValue) {
-      requestHeaders[config.xsrfHeaderName] = xsrfValue;
+      if (typeof config.xsrfHeaderName === 'function') {
+        requestHeaders = config.xsrfHeaderName(xsrfValue, requestHeaders) || requestHeaders;
```

Comment:
It seems like this could be simplified to `config.xsrfHeaderName(xsrfValue, requestHeaders)` since a reference to `requestHeaders` it doesn't need to be reassigned outside the function call. Internally it can add headers as needed.

```js
config.xsrfHeaderName = (val, headers) => {
  headers['foo'] = val;
  headers['bar'] = 'baz';
};
```